src/main/java/core/basesyntax/bookstore/controller/BookController.java

```diff
@@ -29,21 +38,61 @@ public class BookController {
     private static final char PATH_SEPARATOR = '/';
     private final BookService bookService;
 
+    @Operation(summary = "Get all books with pagination")
+    @ApiResponse(
+            responseCode = "200",
+            description = "Found books with pagination"
+    )
     @GetMapping
-    public List<BookDto> getAllBooks() {
-        return bookService.findAll();
+    public List<BookDto> getAllBooks(@ParameterObject Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
+    @Operation(summary = "Get book by id")
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "Found the book",
+                    content = @Content(schema = @Schema(implementation = BookDto.class))
+            ),
+            @ApiResponse(
+                    responseCode = "404",
+                    description = "Didn't find the book",
+                    content = @Content(schema = @Schema(implementation = ExceptionDto.class))
+            )
+    })
     @GetMapping("/{id}")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getById(id);
     }
 
+    @Operation(summary = "Get books with filters and pagination")
+    @ApiResponse(
+            responseCode = "200",
+            description = "Found books",
+            content = @Content(mediaType = "application/json")
+    )
     @PostMapping("/search")
-    public List<BookDto> getBooksBySpec(@RequestBody BookParamsDto params) {
-        return bookService.findAll(params);
+    public List<BookDto> getBooksBySpec(@RequestBody BookParamsDto params,
+                                        @ParameterObject Pageable pageable) {
+        return bookService.findAll(params, pageable);
     }
 
+    @Operation(summary = "Create book")
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "201",
+                    description = "Created new book",
+                    content = @Content(schema = @Schema(implementation = BookDto.class))
+            ),
+            @ApiResponse(
+                    responseCode = "400",
+                    description = "Validation failed",
+                    content = @Content(schema = @Schema(
+                            implementation = ValidationExceptionDto.class
+                    ))
+            )
+    })
     @PostMapping
     public ResponseEntity<?> createBook(@RequestBody @Valid CreateBookRequestDto requestDto,
```

Comment:
Why do you need wrapping to ResponseEntity?

Reply:
response status created + url